hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/NextKSketch.java

```diff
@@ -0,0 +1,153 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.hiero.sketch.dataset.api.ISketch;
+import org.hiero.sketch.dataset.api.PartialResult;
+import org.hiero.sketch.table.*;
+import org.hiero.sketch.table.api.*;
+import rx.Observable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedMap;
+
+public class NextKSketch implements ISketch<ITable, NextKList> {
+    @NonNull
+    private final RecordOrder recordOrder;
+    private final RowSnapshot topRow;
+    private final int maxSize;
+
+    public NextKSketch(@NonNull RecordOrder recordOrder, @NonNull RowSnapshot topRow, int maxSize) {
+        this.recordOrder = recordOrder;
+        this.topRow = topRow;
+        this.maxSize = maxSize;
+    }
+
+    /**
+     * Given a table, generate the Next K items in Sorted Order starting from a specified
+     * rowSnapShot (topRow), together with counts.
+     * @param data The input table on which we want to compute the NextK list.
+     * @return A NextKList.
+     */
+    public NextKList getNextKList(ITable data) {
+        IndexComparator comp = this.recordOrder.getComparator(data);
+        TreeTopK<Integer> topK = new TreeTopK<Integer>(this.maxSize, comp);
+        IRowIterator rowIt = data.getRowIterator();
+        RowTableComparison rowToTable = new RowTableComparison(this.topRow, data, this.recordOrder);
```

Comment:
This code will work if RowTableComparison is ready to accept a null for topRow.
Otherwise you will need to take a different path.